audeer/core/io.py

```diff
@@ -351,39 +362,35 @@ def extract_archive(
     )
     disable = not verbose
 
+    def extract_zip(archive: str) -> list:
+        with zipfile.ZipFile(archive, "r") as zf:
+            members = zf.infolist()
+            for member in progress_bar(members, desc=desc, disable=disable):
+                zf.extract(member, destination)
+            return [m.filename for m in members]
+
+    def extract_tar(archive: str) -> list:
+        with tarfile.open(archive, "r") as tf:
+            members = tf.getmembers()
+            for member in progress_bar(members, desc=desc, disable=disable):
+                # In Python 3.12 the `filter` argument was introduced,
```

Comment:
The value of the filter arg we have discussed previously, and it needn't be reiterated again in this MR.  Please correct me if I'm wrong here.

Reply:
Then this can be resolved and it needn't be resumed here.